scripts/purge_queue.py

```diff
@@ -0,0 +1,83 @@
+import ast
+import sys
+
+import argparse
+import boto3
+
+from azul import config
+
+parser = argparse.ArgumentParser(description=__doc__)
+
+subparsers = parser.add_subparsers(help="sub-command help")
+
+parser_dump = subparsers.add_parser("dump", help="Dump contents of queue into designated file")
+parser_dump.add_argument("queue_name", help="Name of the queue to obtain the messages from")
+parser_dump.add_argument("file_name", help="Name of file to direct output")
+parser_dump.add_argument("--delete", "-d", action='store_true', help="Purge messages from the queue")
+
+parser_list = parser.add_argument("--list", "-l", help="List all available queues", action="store_true")
+
+parser_json_body = parser.add_argument("--json-body", "-jb", help="Output messages in JSON format", action="store_true")
+
+
+def get_messages_from_queue(args):
+    global output_file_name, queue_name
+    sqs = boto3.resource("sqs")
+
+    if args.list:
+        prefix = config._index_prefix  # Limit queues displayed to our deployment prefix
+        for queue_item in list(filter(lambda queue_url: prefix in queue_url, [q.url for q in sqs.queues.all()])):
+            print(queue_item.split("/")[-1])  # Last element in array is the exact queue name!
+        sys.exit(0)
+    elif args.queue_name or args.file_name:  # If one is missing `argparse` would handle it before getting here
+        queue_name = sqs.get_queue_by_name(QueueName=args.queue_name).url
+        output_file_name = args.file_name
+    Queue = sqs.Queue(url=queue_name)
+
+    seconds = 60
+    minutes = 5
+    list_of_messages = []
+    list_of_entries = []
+    while True:
+
+        response = Queue.receive_messages(
+            AttributeNames=['All'],
+            MaxNumberOfMessages=10,
+            VisibilityTimeout=seconds * minutes
+        )
+
+        if len(response) == 0:  # Nothing left in queue
+            break
+        else:
+            for Message in response:
+                entry = {"MessageId": Message.message_id,
+                         "ReceiptHandle": Message.receipt_handle,
+                         "MD5OfBody": Message.md5_of_body,
+                         "Body": (ast.literal_eval(Message.body) if args.json_body else Message.body),
+                         "Attributes": Message.attributes}
+                list_of_entries.append(entry)
+            list_of_messages.append(response)
+
+    with open(output_file_name, "w") as f:
+        input = {"queue": queue_name, "messages": list_of_entries}
+        f.write(str(input))
+    manipulate_queue(Queue=Queue, messages_list=list_of_messages, delete=args.delete)
+    return sys.exit("File : '{}' written success!".format(output_file_name))
+
+
+def manipulate_queue(Queue, messages_list, delete=False):
```

Comment:
Don't couple the batches in which you receive messages to the batches in which you delete or return them. I can explain why in person. Collect all received messages in a simple list of messages. Then use https://more-itertools.readthedocs.io/en/stable/api.html#more_itertools.chunked to split them into batches. I recommend the use of `messages`, `message_batch` and `message_batches` for the variable / argument names.